### PR TITLE
dns: use default dns refresh ttl when DNS returns an NXDOMAIN

### PR DESCRIPTION
**Pull request by crazyxy**
Signed-off-by: Yan Xue <yxyan@google.com>

Description: Use default DNS refresh TTL when DNS returns an NXDOMAIN
Risk Level: Medium
Testing: unit test, manual test
Docs Changes: N/A
Release Notes: N/A
Fixes #Issue: https://github.com/envoyproxy/envoy/issues/7808

### CODE REVIEW

File: source/common/upstream/strict_dns_cluster.cc [hunk 1]
```diff
@@ -138,8 +138,10 @@ void StrictDnsClusterImpl::ResolveTarget::startResolve() {
 
         std::chrono::milliseconds final_refresh_rate = parent_.dns_refresh_rate_ms_;
 
-        if (parent_.respect_dns_ttl_ && ttl_refresh_rate != std::chrono::seconds(0)) {
+        if (parent_.respect_dns_ttl_ && ttl_refresh_rate != std::chrono::seconds(0) &&
```
Comment (htuch) [inline source/common/upstream/strict_dns_cluster.cc]: Do we also need this for logical DNS? Otherwise LGTM.

Comment (crazyxy) [reply source/common/upstream/strict_dns_cluster.cc]: No, we don't need this for logical DNS. If no records returns, dns refresh ttl will be set to default value (5000ms). The reason why strict dns fails is that the init value is set to `std::chrono::seconds::max`.

**Comment by htuch** [comment]
@crazyxy can you merge master to fix some unrelated CI issues?

**Pull request merged**